src/comparator/message_comparator.py

```diff
@@ -191,7 +191,7 @@ def _compare_resources(
 
                 # If there is pattern removal, or pattern value change. Then the global file-level resource
                 # can not replace the original message-level resource.
-                if self._compatible_patterns(
+                if not self._compatible_patterns(
```

Comment:
The `compatible_patterns` returns true if the two patterns are compatible, else return false.

Reply:
Is there a test for `_compatible_patterns`?

Reply:
unit test for this method is not existing, I specifically added it to the `test_message_comparator` to make sure it's working as expected. Thanks!